src/installer/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -233,6 +239,19 @@ void FindBundleHeader()
             return headerOffset != 0;
         }
 
+        private static void CodeSign(string args)
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                return;
+
+            const string codesign = @"/usr/bin/codesign";
+            if (!File.Exists(codesign))
+                return;
+
+            using (var p = Process.Start(codesign, args))
+                p.WaitForExit();
```

Comment:
I think we should check the result - what if signing fails for whatever reason?
Can we make it fail to test this? (maybe give it a nonMACHO input?)

Ideally this would print out the error from the tool (or there would at least be a way to get that) and also fail the build.

Please also consider making codesigning optional even on mac - maybe just an internal MSBuild property (`_EnableMacOSCodeSign` or something like that). To have a workaround for cases where this may cause issues.

Another consideration is publish. Since build runs as part of publish, the executable will get signed and then it will be published. What happens if the developer then tries to sign the executable again? This time with a real cert.

Reply:
I changed the approach so we have an argument to enable the signing in macOS. This will be set from within the SDK in the CreateAppHost task.

> Can we make it fail to test this? (maybe give it a nonMACHO input?)

I added a test case in which we sign it two times, causing codesign to fail, thus throwing an exception with the error printed by the tool. The idea here is to catch the exception in the CreateAppHost and use the SDK logging mechanism for printing the error.

> What happens if the developer then tries to sign the executable again? This time with a real cert.

If signing with `codesign -s <cert> <executable>` it will complain that the app was already sign; this can be worked around by forcing the signing (`codesign -s <cert> -f <executable>`).